src/Agent.Plugins/PipelineArtifact/BuildServer.cs

```diff
@@ -84,5 +84,13 @@ public BuildServer(VssConnection connection)
         {
             return _buildHttpClient.GetArtifactsAsync(project, pipelineId, userState: null, cancellationToken: cancellationToken);
         }
+
+        public Task<List<BuildDefinition>> GetDefinitionsAsync(
+            Guid project,
+            string definitionName,
+            CancellationToken cancellationToken)
+        {
+            return _buildHttpClient.GetFullDefinitionsAsync(project, definitionName, cancellationToken: cancellationToken);
```

Comment:
Was there a reason you choose GetFullDefinitionsAysnc, instead of using GetDeifnitions - https://github.com/microsoft/azure-pipelines-tasks/blob/c014c185b7b62ab42d466a4f0de6316f2909e0c2/Tasks/DownloadBuildArtifactsV0/main.ts#L148 , similar to FCS task?

Reply:
I think we should make it async,  please add async...await..

Reply:
I've tried the GetDeifnitions and in the build client, it needs the def id to get the def info, which doesn't fit our need.